### PR TITLE
CORE-5657 Introduced Verification Sandbox Type

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- add `Verification` as a Sandbox type (it was already added: https://github.com/corda/corda-runtime-os/blob/d83908a3c36a44bced29936609aa6c32bc05310a/libs/virtual-node/sandbox-group-context/src/main/kotlin/net/corda/sandboxgroupcontext/SandboxGroupType.kt#L8)
- ensure that when this type is chosen, SandboxGroupContextService.getOrCreate() can return such sandbox
- add security permissions for VERIFICATION sandbox (same as for FLOW sandbox)

### CODE REVIEW

**Comment by mbrkic-r3** [comment]
Removed comments from policy files as they grew too big and hard to read

File: components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VirtualNodeService.kt [hunk 1]
```diff
@@ -63,9 +63,9 @@ class VirtualNodeService @Activate constructor(
         }
     }
 
-    fun loadSandbox(resourceName: String): SandboxGroupContext {
+    fun loadSandbox(resourceName: String, type: SandboxGroupType = SandboxGroupType.FLOW): SandboxGroupContext {
```
Comment (barrylapthorn) [inline components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VirtualNodeService.kt]: Can this be changed to not having `FLOW` as a default value, and changing the call site(s) to explicitly use `FLOW` instead?

Or is that fine ^^^?  Personally, I don't like implicit default parameters.

File: components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VerificationSandboxTest.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    fun `retrieving environment fails when permission denied`() {
+        securityManagerService.denyPermissions("VERIFICATION/*", listOf(
+            RuntimePermission("getenv.*", null)
+        ))
+
+        val sandboxGroupContext = virtualNode.loadSandbox(CPB1, SandboxGroupType.VERIFICATION)
+        assertThrows<AccessControlException> {
+            virtualNode.runFlow<Map<String, String>>(CPK1_ENVIRONMENT_FLOW, sandboxGroupContext)
+        }
+    }
```
Comment (barrylapthorn) [inline components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VerificationSandboxTest.kt:86]: Does this need a side-by-side test showing the permission setting works in one sandbox but not in the other, i.e. 

```


        val other = virtualNode.loadSandbox(CPB1, SandboxGroupType.FLOW)
        
        virtualNode.runFlow<Map<String, String>>(CPK1_ENVIRONMENT_FLOW, other)
        
        // does not throw
```

**Review by barrylapthorn — COMMENTED** [review]
Just a couple of queries...

File: components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VirtualNodeService.kt (hunk 1 above)
Comment (chrisr3) [reply components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VirtualNodeService.kt]: A default value made sense back when `FLOW` was the _only_ type of sandbox. Now, not so much.... :grin:

Comment (mbrkic-r3) [reply components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VirtualNodeService.kt]: changed

File: components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VerificationSandboxTest.kt (hunk 2 above)
Comment (mbrkic-r3) [reply components/virtual-node/sandbox-group-context-service/src/integrationTest/kotlin/net/corda/sandboxgroupcontext/test/VerificationSandboxTest.kt:86]: added

**Review by barrylapthorn — APPROVED** [review]
Seems reasonable.

**Review by chrisr3 — APPROVED** [review]
Looks OK, although those policy files aren't being displayed in my browser.

**Pull request merged**